src/course-updates/hooks.jsx

```diff
@@ -0,0 +1,114 @@
+import { useDispatch, useSelector } from 'react-redux';
+import moment from 'moment/moment';
+import { useEffect, useState } from 'react';
+import { useToggle } from '@edx/paragon';
+
+import { getCourseHandouts, getCourseUpdates } from './data/selectors';
+import { REQUEST_TYPES } from './constants';
+import { COMMA_SEPARATED_DATE_FORMAT } from '../constants';
+import {
+  createCourseUpdateQuery,
+  deleteCourseUpdateQuery,
+  editCourseHandoutsQuery,
+  editCourseUpdateQuery,
+  fetchCourseHandoutsQuery,
+  fetchCourseUpdatesQuery,
+} from './data/thunk';
+
+const useCourseUpdates = ({ courseId }) => {
+  const dispatch = useDispatch();
+  const initialUpdate = { id: 0, date: moment().utc().toDate(), content: '' };
+
+  const [requestType, setRequestType] = useState('');
+  const [isUpdateFormOpen, openUpdateForm, closeUpdateForm] = useToggle(false);
+  const [isDeleteModalOpen, openDeleteModal, closeDeleteModal] = useToggle(false);
+  const [currentUpdate, setCurrentUpdate] = useState(initialUpdate);
+
+  const courseUpdates = useSelector(getCourseUpdates);
+  const courseHandouts = useSelector(getCourseHandouts);
+
+  const courseUpdatesInitialValues = requestType === REQUEST_TYPES.edit_handouts
```

Comment:
Can we use destructuring for `REQUEST_TYPES`? This will shorten the code a bit and make it more readable.

`const { 
        edit_handouts, 
        add_new_update,  
        edit_update, 
        delete_update, 
        etc... } = REQUEST_TYPES;`